lib/bundler/gem_version_promoter.rb

```diff
@@ -129,16 +129,11 @@ def sort_dep_specs(spec_groups, locked_spec)
           b_ver <=> a_ver
         when !minor? && segments_do_not_match(:minor, a_ver, b_ver)
           b_ver <=> a_ver
+        when !unlocking_gem?(gem_name) && one_version_matches(locked_version, a_ver, b_ver)
```

Comment:
part of what i don't like about this is passing a_ver and b_ver around all over (OOP smell). I have similar code in bundler-patch that I tried recently making ivars of these (@a_ver and @b_ver) so I don't have to pass them through to see if I like it -- then it just reads `one_version_matches(locked_version)` -- what would you think about that?